### PR TITLE
Support odometry velocity in body and local frame

### PR DESCRIPTION
**Pull request by kamilritz**
At the moment we expect the external vision's velocity to be expressed in world/local frame. This PR allows for the vision velocity also being expressed in body frame. For this a field is added to the vehicle_odometry message that stores the frame in which the velocity is expressed;

**Related ECL PR:** https://github.com/PX4/ecl/pull/708
**Solving issue:** https://github.com/PX4/Firmware/issues/13935

FYI @nicovanduijn @bresch @jkflying

### CODE REVIEW

File: src/modules/simulator/simulator_mavlink.cpp [hunk 1]
```diff
@@ -1004,10 +993,12 @@ int Simulator::publish_odometry_topic(const mavlink_message_t *odom_mavlink)
 			odom.pose_covariance[i] = odom_msg.pose_covariance[i];
 		}
 
-		/* The velocity in the local NED frame */
-		odom.vx = linvel_local(0);
-		odom.vy = linvel_local(1);
-		odom.vz = linvel_local(2);
+		/* The velocity in the body-fixed frame */
+		odom.velocity_frame = odom.BODY_FRAME_FRD
+				      odom.vx = odom_msg.vx;
```
Comment (TSC21) [inline src/modules/simulator/simulator_mavlink.cpp]: @kamilritz looks like wrong indentation.

File: src/modules/simulator/simulator_mavlink.cpp [hunk 2]
```diff
@@ -1004,10 +993,12 @@ int Simulator::publish_odometry_topic(const mavlink_message_t *odom_mavlink)
 			odom.pose_covariance[i] = odom_msg.pose_covariance[i];
 		}
 
-		/* The velocity in the local NED frame */
-		odom.vx = linvel_local(0);
-		odom.vy = linvel_local(1);
-		odom.vz = linvel_local(2);
+		/* The velocity in the body-fixed frame */
+		odom.velocity_frame = odom.BODY_FRAME_FRD
```
Comment (dagar) [inline src/modules/simulator/simulator_mavlink.cpp]:
```suggestion
		odom.velocity_frame = odom.BODY_FRAME_FRD;
```

**Comment by kamilritz** [comment]
NOTE: Last testflights after rebase showed strange behavior. But could also be the vehicle itself. Please fly test this again before merging.

**Comment by TSC21** [comment]
@kamilritz are you able to rebase and take this further?

**Comment by kamilritz** [comment]
@TSC21 Rebasing is not a problem, but I am not able to test it myself.

**Comment by TSC21** [comment]
@kamilritz ok rebase should be enough for now. Can you do that? I will sync with @jkflying regarding how this can be tested.

**Comment by kamilritz** [comment]
@TSC21 Ok, I will start refactoring as soon as we have a solution for the testing.

**Comment by jkflying** [comment]
@TSC21 is there some way we could do ground-truth VIO in SITL?

I'm happy with this PR, but we knew it was working before and there is no test coverage, so I don't want to accidentally break something.

**Comment by TSC21** [comment]
> @TSC21 is there some way we could do ground-truth VIO in SITL?
> 
> I'm happy with this PR, but we knew it was working before and there is no test coverage, so I don't want to accidentally break something.

@jkflying we seemed to have that before but it looks like it was removed - https://github.com/PX4/Firmware/commit/6a4a4bd0d69aa19d69577f72aa3cde8c846ced98#diff-436758762aea1e43c9654faf66315930L42-L43.

**Comment by kamilritz** [comment]
I think there still is some basic testing: https://github.com/PX4/Firmware/blob/master/test/mavsdk_tests/configs/sitl.json#L18

**Comment by jkflying** [comment]
Considering the current issues I'm happy to merge this based on the SITL tests. I don't think it's realistic to get better testing than that for now.

**Comment by kamilritz** [comment]
I just remember that the last test I did on a real platform, did not went well. And there was not enough time left to investigate why. Based on that I am a bit hesitant to get this in without further testing a real platform.

**Comment by jkflying** [comment]
We found that this was due to a problem on the vehicle, so that doesn't give any information to whether this PR works or not :slightly_smiling_face:

**Comment by kamilritz** [comment]
Nice. I will rebase it then.

**Comment by kamilritz** [comment]
Rebased the PR and reopened it here: https://github.com/PX4/Firmware/pull/14703

**Pull request closed without merging**